### PR TITLE
Bloom Pass - Inspired from Unreal Engine

### PR DESCRIPTION
**Pull request by spidersharma03**
This Pass consists of the following passes inside::

Extract Bright Pass
Mip Map blur Pass
Combine Pass

### CODE REVIEW

**Comment by bhouston** [comment]
Here is the results:

http://ci.threejs.org/api/pullrequests/9351/examples/webgl_postprocessing_unreal_bloom.html

image

**Comment by bhouston** [comment]
Mrdoob reported that on mobile this shader isn't working well: https://twitter.com/mrdoob/status/755144086844153856  I believe he is running with a Nexus 5.

**Comment by spidersharma03** [comment]
Fixed the issue of weird size on mobile.

**Comment by mrdoob** [comment]
Thanks!

File: src/renderers/shaders/ShaderChunk/common.glsl [hunk 1]
```diff
@@ -72,3 +72,9 @@ vec3 linePlaneIntersect( in vec3 pointOnLine, in vec3 lineDirection, in vec3 poi
 	return lineDirection * ( dot( planeNormal, pointOnPlane - pointOnLine ) / dot( planeNormal, lineDirection ) ) + pointOnLine;
 
 }
+
+float gaussianPdf(in float x, in float sigma) {
```
Comment (mrdoob) [inline src/renderers/shaders/ShaderChunk/common.glsl]: @bhouston How about moving this to `UnrealBloomPass` shader instead?

**Pull request merged**